### ISSUE TITLE
ArgumentError when upgrading to 1.3.3

### ISSUE THREAD

**Issue by jcoyne** [dry-rb/dry-schema#186]
**Describe the bug**
I was using 1.3.2 and my code was working, when I upgraded to 1.3.3 I started getting this error:
```
     Failure/Error: required(:service_conforms_to) { array? { each(:str?) } }

     ArgumentError:
       wrong number of arguments (given 0, expected 1)
```

**To Reproduce**

I have a schema like this:

```ruby

ServicesSchema = Dry::Schema.Params do
  required(:service_conforms_to) { array? { each(:str?) } }
end
```

**Expected behavior**

No ArgumentError when upgrading past 1.3.2

**Your environment**

- Affects my production application: no
- Ruby version: 2.6.4
- OS: OsX/Linux

**Comment by jcoyne** [comment]
I was able to work around this by switching to `required(:service_conforms_to).array(:str?)` syntax, but it was surprising for the API to change in a patch version.

**Comment by solnic** [comment]
This is a bug. The API was not supposed to change. I'll investigate. Thanks for reporting it. FWIW your solution is something I would recommend over block-based syntax anyway. The only reason why block-based syntax should be used is for complex cases like using disjunctions etc.

**Comment by flash-gordon** [comment]
https://github.com/dry-rb/dry-schema/blob/bdcfa4f6af789a65632572d468e4351df00975c9/lib/dry/schema/macros/value.rb#L43 ^ this was broken here, `array?` wasn't the right choice for the method, apparently.

**Comment by solnic** [comment]
This is now fixed in 1.3.4.
